### PR TITLE
Max requests

### PR DESCRIPTION
**Pull request by whites11**
Towards: https://github.com/giantswarm/giantswarm/issues/23532 and https://github.com/giantswarm/giantswarm/issues/22234#issuecomment-1141952289

This PR automatically sets the `--max-requests-inflight`, `--max-mutating-requests-inflight` and resource limits to API Server's manifest based on node size.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: pkg/template/master_template.go [hunk 1]
```diff
@@ -111,6 +111,26 @@ systemd:
...
+      Type=oneshot
+      RemainAfterExit=yes
+      TimeoutStartSec=0
+      Environment=IMAGE={{ .Images.Envsubst }}
+      ExecStartPre=/opt/bin/setup-apiserver-environment
+      ExecStart=docker run --rm \
+        --env-file /etc/apiserver-environment \
+        -v /etc/kubernetes/:/etc/kubernetes/ \
+        $IMAGE \
+        ash -c "cat /etc/kubernetes/manifest-templates/k8s-api-server.yaml.tmpl |envsubst >/etc/kubernetes/manifests/k8s-api-server.yaml"
```
Comment (njuettner) [inline pkg/template/master_template.go:131]: Typo?

```suggestion
        sh -c "cat /etc/kubernetes/manifest-templates/k8s-api-server.yaml.tmpl |envsubst >/etc/kubernetes/manifests/k8s-api-server.yaml"
```

Comment (njuettner) [reply pkg/template/master_template.go:131]: Also `/etc/kubernetes/manifest-templates/k8s-api-server.yaml.tmpl` is `/etc/kubernetes/manifest/k8s-api-server.yaml.tmpl`

**Review by njuettner — APPROVED** [review]

File: pkg/template/master_template.go (hunk 1 above)
Comment (whites11) [reply pkg/template/master_template.go:131]: nope, `ash` is `alpine shell` and it has a bit more features than plain `sh`

Comment (njuettner) [reply pkg/template/master_template.go:131]: Ah right 👍🏻

**Pull request merged**